### PR TITLE
Integrate extended diagnostics (#1300 with #1295)

### PR DESCRIPTION
**Pull request by aryoda**
Integrates:
* #1295
* #1300

Changes:
* Add user-callback path to collect_diagnostics() to avoid regression
* Remove TravisCI test failure workaround (was fixed with #1305)
* Correct some typos in comments.

### CODE REVIEW

**Comment by aryoda** [comment]
The output is now:

```
> ./backintime --diagnostics
{
    "backintime": {
        "name": "Back In Time",
        "version": "1.3.2",
        "config-version": 6,
        "distribution-package": "/home/UsernameReplaced/dev/backintime",
        "started-from": "/home/UsernameReplaced/dev/backintime/common",
        "running_as_root": false,
        "user_callback": "/home/UsernameReplaced/.config/backintime/user-callback",
        "git-branch": "PR/integrate_extended_diagnostics",
        "git-hash": "19c3d25a07a74d4f1023a1c7b1cf8d2d97e5408c"
    },
    "host-setup": {
        "platform": "Linux-5.15.0-48-generic-x86_64-with-glibc2.29",
        "system": "Linux #54~20.04.1-Ubuntu SMP Thu Sep 1 16:17:26 UTC 2022",
        "os-release": "Ubuntu 20.04.5 LTS",
        "display-system": "x11",
        "locale": "en_US, UTF-8",
        "PATH": "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"
    },
    "python-setup": {
        "python": "3.8.10 default Jun 22 2022 20:18:18 CPython GCC 9.4.0",
        "sys.path": [
            "/home/UsernameReplaced/dev/backintime/qt/plugins",
            "/home/UsernameReplaced/dev/backintime/common/plugins",
            "/home/UsernameReplaced/dev/backintime/plugins",
            "/home/UsernameReplaced/dev/backintime/common",
            "/usr/lib/python38.zip",
            "/usr/lib/python3.8",
            "/usr/lib/python3.8/lib-dynload",
            "/usr/local/lib/python3.8/dist-packages",
            "/usr/lib/python3/dist-packages"
        ],
        "qt": "PyQt 5.14.1 / Qt 5.12.8"
    },
    "external-programs": {
        "rsync": "3.1.3",
        "ssh": "OpenSSH_8.2p1 Ubuntu-4ubuntu0.5, OpenSSL 1.1.1f  31 Mar 2020",
        "sshfs": "2.10.0",
        "encfs": "(no encfs)",
        "shell": "/bin/bash",
        "shell-version": "GNU bash, version 5.0.17(1)-release (x86_64-pc-linux-gnu)"
    }
}
```

**Comment by aryoda** [comment]
Just saw that
- `user_callback` uses an underscore instead of hyphen
- `running_as_root` also uses underscores

Shall I fix this (using hyphens only like everywhere else)?

File: common/diagnostics.py [hunk 1]
```diff
@@ -16,22 +16,23 @@
...
-    The informatinos can be used e.g. for debugging and but reports.
+    The information can be used e.g. for debugging and bug reports.
 
     Returns:
        dict: A nested dictionary.
     """
     result = {}
 
-    USER_REPLACED = 'UsernameReplaced'
+    USER_REPLACED = 'UsernameReplaced'  # replace home folder user names with this dummy name for privacy reasons
```
Comment (buhtzz) [inline common/diagnostics.py:29]: Violates PEP8. Line is to long.

```
        # Replace home folder user names with this dummy name for
        # privacy reasons.
        USER_REPLACED = 'UsernameReplaced'
```

File: common/diagnostics.py [hunk 2]
```diff
@@ -16,22 +16,23 @@
...
 
     pwd_struct = pwd.getpwuid(os.getuid())
 
     # === BACK IN TIME ===
     distro_path = _determine_distro_package_folder()
+    cfg = config.Config()    # work-around: Instantiate to get the user-callback folder (should be singleton)
```
Comment (buhtzz) [inline common/diagnostics.py:35]:
```
        # work-around: Instantiate to get the user-callback folder
        # (should be singleton)
        cfg = config.Config()
```

**Review by buhtzz — CHANGES_REQUESTED** [review]
Moin,
fine work.

I'm just becoming warm with the GitHub review front-end. I wasn't able to directly edit the files. Maybe because of user rights.
So I added my code change suggestions as comments.

**Comment by aryoda** [comment]
@buhtzz THX for the reviews. Findings fixed (rebased), now it's @emtiu turn...

**Review by buhtzz — APPROVED** [review]

**Comment by aryoda** [comment]
I have added another commit to

- add the local and global config file info
- rename "config-version" to "latest-config-version" (to avoid any confusion with the actual config version in existing config files...)

**Review by buhtzz — APPROVED** [review]

**Comment by emtiu** [comment]
Thanks for the good work, and sorry for being slow to merge ;)

**Pull request merged**